libbeat/docs/processors-config.asciidoc

```diff
@@ -453,3 +454,67 @@ section.
 NOTE: If you define an empty list of fields under `include_fields`, then only
 the required fields, `@timestamp` and `type`, are exported.
 
+[[kubernetes]]
+=== kubernetes
+
+The `kubernetes` processor annotates each event with relevant metadata based on
+which kubernetes pod the event originated from. Each event is annotated with:
+
+* Pod Name
+* Namespace
+* Labels
+
+The `kuberentes` process has two basic building blocks which are:
+
+* Indexers
+* Matchers
+
+Indexers take in a pod's metadata and builds indices based on the pod metadata. For example,
+the `ip_port` indexer can take a kubernetes pod and index the pod metadata based on all
+`pod_ip:container_port` combinations.
+
+Matchers are used to build lookup keys with which indices can be queried. For example, the `fields`
+matcher can take a list of fields based on whose values, indices can be queried to retrieve metadata.
+
+Each beat can define its own default indexers and matchers which are enabled by default. For example,
+filebeat enables the `container` indexer which indexes pod metadata based on all container IDs and a
+ `logs_path` matcher that takes the `source` field, extracts the container ID and uses it to retrieve
+ metadata.
+
+ The configuration below enables the processor when the beat as a pod in kubernetes.
+
+[source,yaml]
+-------------------------------------------------------------------------------
+processors:
+- kuberenes:
+    in_cluster: true
+-------------------------------------------------------------------------------
+
+The configuration below enables the processor on a beat running as a process on the kubernetes node
+[source,yaml]
+-------------------------------------------------------------------------------
+processors:
+- kuberenes:
+    in_cluster: false
+    host: <hostname>
+    kube_config: ~/.kube/config
+-------------------------------------------------------------------------------
+
+The configuration below allows custom indexers and matchers to be enabled.
+
+[source,yaml]
+-------------------------------------------------------------------------------
+processors:
+- kuberenes:
+    in_cluster: false
+    host: <hostname>
+    kube_config: ~/.kube/config
+    enable_default_indexers: false
+    enable_default_matchers: false
+    indexers:
+      - ip_port:
+    matchers:
+      - fields:
+          lookup_fields: ["metricset.host"]
+-------------------------------------------------------------------------------
```

Comment:
I miss a note on how where the enriched data for the event will be put. How will the end event look like?